components/reference-data/src/main/java/org/datacleaner/beans/transform/RemoveDictionaryMatchesTransformer.java

```diff
@@ -98,27 +134,62 @@ public void close() {
     @Override
     public Object[] transform(InputRow inputRow) {
         final String value = inputRow.getValue(_column);
-        final String result = transform(value);
-        return new Object[] { result };
+        final Object[] result = transform(value);
+        return result;
     }
 
-    public String transform(final String value) {
-        if (Strings.isNullOrEmpty(value)) {
-            return value;
+    public Object[] transform(final String value) {
+        switch (_removedMatches) {
+        case LIST:
+            return transform2(value);
+        default:
+            return transformDefault(value);
```

Comment:
The logic for the first value (value without dictionary matches) is now duplicated code. Wouldn't it be better to make it a shared codebase and just handle a match-list differently. That way we don't branch out at the top level of the transform-method but just have two (maybe more) different ways of dealing with the "Removed matches".